### DISCUSSION TITLE
LCD screen displaying random characters or just black squares

### DISCUSSION THREAD

**Discussion by dlawle** [probonopd/MiniDexed#199] [Q&A] [answered]
Hello, 

I’m running into an issue where upon the first connection, my lcd display showed this:
![lcd display showed this.](https://user-images.githubusercontent.com/39495124/166393210-e0007268-0af6-4409-b430-5a65762a4525.jpeg)

After a power cycle, it only displayed a single row (top) of black squares (all matrix grid filled). 

I'm running on a pi 4, with a 128gb sd card, followed the tutorial in the linked video. I’ve tried a pot between VO and that didn’t display anything either. Could this be my pi, or could my sd card be too large?

**Comment by dlawle** [comment]
I believe this may belong in discussion. Please move if so!

**Comment by craigyjp** [comment]
Try powering the display from  3.3v.

**Reply by dlawle** [reply]
Will give this a try!

**Reply by craigyjp** [reply]
When I ran my OLED display from 5v I had the same problems.

**Comment by probonopd** [comment]
Also, I find that Dupont cables are not the most reliable. Soldering directly to the solder points of the display may result in a more stable display.

**Reply by dlawle** [reply]
I did fire up the same sd card on a pi zero w, to the same effect.

**Reply by dlawle** [reply]
I’ll give direct soldering a try

**Comment by apvanzanten** [comment]
Hello,

The output you're seeing is definitely not random, I have the exact same output on my display!

I am using a Raspberry Pi Model 2B. Also, my particular model display won't run at all when powered at 3.3v, so that is not an option/solution.

Something I noticed when writing a driver for this exact display a few months ago, that this particular unit is a bit sluggish to respond compared to others I have worked with (i.e. time required with data on bus and enable pin high is long). It might be completely unrelated but, I'm curious. Is there a way that I could tweak the delay times on the driver to experiment/investigate?

**Reply by ehisforadam** [reply]
I am getting the exact same thing with mine. I wonder if there is some difference in hardware from different suppliers that are still technically "compatible."

Edit1: I tested the display with an Ardunio and it worked fine. I tried hooking it to another RasPi and got the same thing. It's gotta be something with how some of these displays work with RasPis.

Edit2 : The pins are wrong in the fritzing diagram. GPIO Pin 11 should go to E (enable on the display) and Pin 13 should be Register Select (SE). On the diagram RS goes to Pin 7. (At least for Pi 3B.) I found this by looking at the minidexed.ini file. It lists all the GPIO pin outs.

**Reply by probonopd** [reply]
According to the recent pin definition on README.md, 
* Raspberry Pi GPIO17 (pin 11) needs to go to LCD EN (pin 6)
* Raspberry Pi GPIO4 (pin 7) needs to go to LCD RS (pin 4)

So, the Fritzing diagrams on https://github.com/probonopd/MiniDexed/wiki/Hardware seem correct.
Could it be that you are confusing the GPIO numbers with the pin numbers?

**Reply by dlawle** [reply]
Interesting. I’ve just seen this change. I’ll give it a test tonight. Do I need to update the base image as well? 

I’m not sure if there’s any real differences/changes but I feel as if I followed the pin assignment to a t. The visual guide will definitely help!

**Reply by ehisforadam** [reply]
I didn't change any of the files, I just swapped the pins around. It takes 10-15 seconds for the screen to change from a row of black boxes to showing the proper text. It looks like I just have an older version of the files that changed between when I downloaded it and started putting everything together. I used the latest info on the pinouts here. I didn't realize changes were so rapidly on this project.

**Comment by dlawle** [comment, marked as answer]
Looks like using the correct pin layout did the trick. Thanks for the help everyone!

**Comment by probonopd** [comment]
With our brand new i2c display support, things should be a lot easier thanks to @diyelectromusic :+1:
